### PR TITLE
JsonUnit has been added

### PR DESCRIPTION
**Pull request by viclovsky**
[//]: # (
. Thank you so much for sending us a pull request! 
.
. Make sure you have a clear name for your pull request. 
. The name should start with a capital letter and no dot is required in the end of the sentence.
. To link the request with isses use the following notation: (fixes #123, fixes #321\)
.
. An example of good pull request names:
. - Add Russian translation (fixes #123\)
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

### Context
Allure integration for JsonUnit has been added. 
JsonPatchMatcher is extension of JsonUnit matcher, that generates pretty html attachment for differences based on [json diff patch](https://github.com/benjamine/jsondiffpatch/blob/master/docs/deltas.md).

Example was added. Screenshot is presented below.
allure report 2018-07-24 01-55-13
allure report 2018-07-24 01-57-00

Also added `withRequestTemplate` and `withResponseTemplate` to customise templates for rest-assured.

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: allure-jsonunit/build.gradle [hunk 1]
```diff
@@ -0,0 +1,14 @@
+description = 'Allure JsonUnit'
+
+apply from: "${gradleScriptDir}/maven-publish.gradle"
+apply from: "${gradleScriptDir}/bintray.gradle"
+apply plugin: 'maven'
+
+dependencies {
+    compile project(':allure-attachments')
+    compile('net.javacrumbs.json-unit:json-unit:2.0.0.RC1')
+    compile('com.google.code.gson:gson')
```
Comment (baev) [inline allure-jsonunit/build.gradle]: we have jackson in deps, can we use it instead?

File: allure-jsonunit/src/main/resources/tpl/diff.ftl [hunk 2]
```diff
@@ -0,0 +1,311 @@
+<html>
+<head>
+    <meta http-equiv = "content-type" content = "text/html; charset=UTF-8">
+    <script type='text/javascript' src="https://cdn.jsdelivr.net/npm/jsondiffpatch/dist/jsondiffpatch.umd.min.js" charset="utf-8"></script>
```
Comment (baev) [inline allure-jsonunit/src/main/resources/tpl/diff.ftl]: lets include minified version directly to template in order to make it work offline

File: allure-jsonunit/src/main/resources/tpl/diff.ftl [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+        margin-left: 30px;
+        display: inline-block;
+    }
+
+    #delta-panel-visual {
+        width: 100%;
+        overflow: auto;
+    }
+
+    #visualdiff {
```
Comment (baev) [inline allure-jsonunit/src/main/resources/tpl/diff.ftl]: use class selectors instead if possible

**Comment by eroshenkoam** [comment]
ty! 👍

**Pull request merged**